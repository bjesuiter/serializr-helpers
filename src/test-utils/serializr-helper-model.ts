import {list, primitive, serializable} from "serializr";

export const demoProp: string = 'Bob der Baumeister';

export const demoArray: string[] = [
    'Lilly',
    'Gunnar',
    'Fjordojan'
];

export class SerializrHelperModel {

    @serializable(primitive())
    demoProp: string = demoProp;

    @serializable(list(primitive()))
    demoArray: string[] = demoArray;

}
