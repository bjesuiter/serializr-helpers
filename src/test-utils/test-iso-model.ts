import {serializable} from "serializr";
import {Moment} from "moment";
import {MomentSerializationSchema} from "../lib/moment-serialization-schema";

export class TestIsoModel {

    @serializable(MomentSerializationSchema())
    public testMoment: Moment;

    constructor(testMoment: Moment) {
        this.testMoment = testMoment;
    }


}
