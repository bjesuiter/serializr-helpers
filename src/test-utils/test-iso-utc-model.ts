import {serializable} from "serializr";
import {Moment} from "moment";
import {MomentSerializationSchema} from "../lib/moment-serialization-schema";

export class TestIsoUtcModel {

    @serializable(MomentSerializationSchema({useUtc: true}))
    public testMoment: Moment;

    constructor(testMoment: Moment) {
        this.testMoment = testMoment;
    }


}
