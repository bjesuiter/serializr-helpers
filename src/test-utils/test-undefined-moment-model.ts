import {serializable} from "serializr";
import {Moment} from "moment";
import {MomentSerializationSchema} from "../lib/moment-serialization-schema";

export class TestUndefinedMomentModel {

    @serializable(MomentSerializationSchema())
    public testMoment?: Moment;

    constructor(testMoment?: Moment) {
        this.testMoment = testMoment;
    }
}
