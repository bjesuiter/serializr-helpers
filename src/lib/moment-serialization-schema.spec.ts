import moment, {Moment} from "moment";
import {TestIsoModel} from "../test-utils/test-iso-model";
import {serialize} from "serializr";

describe('MomentSerializationScheme', function () {

    const testMomentString = '2019-06-10T08:50:23.559+02:00';
    let testIsoMoment: Moment;
    let testIsoModel: TestIsoModel;

    beforeEach(() => {
        testIsoMoment = moment(testMomentString);
        testIsoModel = new TestIsoModel(testIsoMoment);
    });

    describe('MomentIsoSerialization', () => {

        it('serialize', () => {
            const js = serialize(testIsoModel);
            expect(js.testMoment).toEqual(testMomentString);
        });

        it('deserialize', () => {
            const js = JSON.parse(`{
                "testMoment": "${testMomentString}"
            }`);

            expect(js).toEqual(serialize(testIsoModel));
        })

    });

});
