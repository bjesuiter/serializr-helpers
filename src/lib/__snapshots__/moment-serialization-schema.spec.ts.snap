// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MomentSerializationScheme Test buildDeserializer function Test defaultDeserializationValue should throw error when default deserialization value is invalid 1`] = `"Default Moment deserialization value is invalid. Got input string \\"default-illegal-moment\\""`;

exports[`MomentSerializationScheme Test buildDeserializer function Test deserializationErrorPolicies should throw error 1`] = `
"Moment js serialized json value is invalid! 
                    Got the value \\"some-illegal-moment\\" which does not decode into a valid Moment object.
                    You can change the handling of this message by setting the 'deserializationErrorPolicy' in 
                    MomentSerializationOptions"
`;
